include/rtweekend.hpp

```diff
@@ -16,62 +16,66 @@
 #include <build_parameters.hpp>
 #include <xorshift.hpp>
 
+using real_t = float;
+
 // Constants
 
-constexpr float infinity = std::numeric_limits<float>::infinity();
-constexpr float pi = 3.1415926535897932385f;
+constexpr real_t infinity = std::numeric_limits<real_t>::infinity();
+constexpr real_t pi = 3.1415926535897932385f;
```

Comment:
literals are still assuming float which means that if we change real_t to something else we will get some conversions. maybe make our own user-defined literal could help abstract this.

`real_t operator "" _r(long double d) {
    return d;
}`

this would force conversion to be on the literals directly such that they can be constant folded

Reply:
Just use `constinit`